### DISCUSSION TITLE
The difference between setValue() callings

### DISCUSSION THREAD

**Discussion by evgeniyworkbel** [react-hook-form/react-hook-form#10628] [Q&A] [answered]
Hi! I can't see the difference between setValue() callings in [docs](https://react-hook-form.com/api/useform/setvalue/) example. Colud you please explain why in the first case we should specify full path to value, but in the second one we should do it by passing object? Both cases contain nested values
![image](https://github.com/react-hook-form/react-hook-form/assets/93836126/8cbb646d-2427-489a-92b5-07ee538e6e11)

**Comment by evgeniyworkbel** [comment]
Heeelp!

**Comment by leapful** [comment, marked as answer]
@evgeniyworkbel 

The first one is a registered field with name `yourDetails.firstName` and its value is primitive value (a string). If you set an object value for this case, react-hook-form need to walk through via for loop on each prop inside object to compare target prop then set value and it can be less performant. Using directly explicit name `yourDetails.firstName` will be more optimizing as it's now accessed via dictionary prop name.

The second one is a registered field with name `nestedValue` and its value is an object `{ test: 'updatedData' }`. Now the registered name is "nestedValue" therefore `nestedValue.test` will not be a valid registered name inside react-hook-form handling and cannot set value.

**Reply by evgeniyworkbel** [reply]
I caught it. meanwhile cases with nestedValue is rare, isn't it?

**Reply by leapful** [reply]
@evgeniyworkbel 

Yes, I would recommend to dealing with form only via primitive value, since using object value mights lead into potential issues on reference or missing data of the whole object due to side effect.

**Reply by evgeniyworkbel** [reply]
Many thanks!
